core/shared/platform/windows/win_file.c

```diff
@@ -1,115 +1,779 @@
-#include "platform_api_extension.h"
-#include "platform_internal.h"
+#include "bh_common.h"
+#include "wasi_errno.h"
+
+#define CHECK_VALID_WIN_HANDLE(win_handle)                              \
+    do {                                                                \
+        if ((win_handle) == NULL                                        \
+            || ((win_handle)->type == windows_handle_type_socket        \
```

Comment:
given the macro is only used in `win_file.c`, should we rename it to `CHECK_VALID_FILE_HANDLE` and treat socket handle as invalid?

Reply:
Good point - there are couple functions (`os_close`, `os_fstat`) which make sense to call with a socket or file handle so apart from those functions, I'll use a macro `CHECK_VALID_FILE_HANDLE`.